### PR TITLE
Add public docstring to Bio.Seq

### PR DESCRIPTION
**Pull request by Gasta88**
As mentioned in issue #1203 I've added public docstring to `Bio.Seq` module.

### CODE REVIEW

File: Bio/Seq.py [hunk 1]
```diff
@@ -62,7 +63,7 @@ def _maketrans(complement_mapping):
 
 
 class Seq(object):
-    """A read-only sequence object (essentially a string with an alphabet).
+    """Create a read-only sequence object (essentially a string with an alphabet).
```
Comment (peterjc) [inline Bio/Seq.py]: The ``__init__`` could be said to create the object, but that does not make sense for the class docstring. You could just remove the ``A`` and use ``Read-only sequence object (essentially a string with an alphabet).``

File: Bio/Seq.py [hunk 2]
```diff
@@ -198,12 +199,12 @@ def __eq__(self, other):
         return str(self) == str(other)
 
     def __ne__(self, other):
-        """Not equal, see __eq__ documentation."""
+        """Compare sequences with a not-equal operand, see __eq__ documentation."""
```
Comment (peterjc) [inline Bio/Seq.py]: Is this really clearer (likewise for the other special methods)?

File: Bio/Seq.py [hunk 3]
```diff
@@ -513,7 +514,7 @@ def rfind(self, sub, start=0, end=sys.maxsize):
         return str(self).rfind(sub_str, start, end)
 
     def startswith(self, prefix, start=0, end=sys.maxsize):
-        """Does the Seq start with the given prefix?  Returns True/False.
+        """Reurn True if the Seq starts with the given prefix, False otherwise.
```
Comment (peterjc) [inline Bio/Seq.py]: Typo (``Return``)

File: Bio/Seq.py [hunk 4]
```diff
@@ -544,7 +545,7 @@ def startswith(self, prefix, start=0, end=sys.maxsize):
             return str(self).startswith(prefix_str, start, end)
 
     def endswith(self, suffix, start=0, end=sys.maxsize):
-        """Does the Seq end with the given suffix?  Returns True/False.
+        """Return True if the Seq end with the given suffix, False otherwise.
```
Comment (peterjc) [inline Bio/Seq.py]: end --> ends

File: Bio/Seq.py [hunk 5]
```diff
@@ -1139,7 +1140,7 @@ def ungap(self, gap=None):
 
 
 class UnknownSeq(Seq):
-    """A read-only sequence object of known length but unknown contents.
+    """Create a read-only sequence object of known length but unknown contents.
```
Comment (peterjc) [inline Bio/Seq.py]: See comment for the ``Seq`` class docstring, ``Create ...`` is not quite right here.

File: Bio/Seq.py [hunk 6]
```diff
@@ -1391,7 +1394,7 @@ def complement(self):
         return self
 
     def reverse_complement(self):
-        """The reverse complement of an unknown nucleotide equals itself.
+        """Return the reverse complement of an unknown nucleotide equals itself.
```
Comment (peterjc) [inline Bio/Seq.py]: The new version does not make sense to me.

File: Bio/Seq.py [hunk 7]
```diff
@@ -1556,7 +1559,7 @@ def ungap(self, gap=None):
 
 
 class MutableSeq(object):
-    """An editable sequence object (with an alphabet).
+    """Create an editable sequence object (with an alphabet).
```
Comment (peterjc) [inline Bio/Seq.py]: Again, ``Create ...`` is not appropriate.

File: Bio/Seq.py [hunk 8]
```diff
@@ -1986,7 +2001,7 @@ def transcribe(dna):
 
 
 def back_transcribe(rna):
-    """Back-transcribes an RNA sequence into DNA.
+    """Return the RNA back-transcribed sequence into DNA.
```
Comment (peterjc) [inline Bio/Seq.py]: How about ``Return the RNA sequence back-transcribed into DNA.``

File: Bio/Seq.py [hunk 9]
```diff
@@ -2009,7 +2024,7 @@ def back_transcribe(rna):
 
 def _translate_str(sequence, table, stop_symbol="*", to_stop=False,
                    cds=False, pos_stop="X", gap=None):
-    """Helper function to translate a nucleotide string (PRIVATE).
+    """Translate a nucleotide to string (PRIVATE).
```
Comment (peterjc) [inline Bio/Seq.py]: "a nucleotide" implies a single base letter.

How about ``Translate nucleotide string into a protein string (PRIVATE).``

File: Bio/Seq.py (hunk 2 above)
Comment (Gasta88) [inline Bio/Seq.py]: How about `"""Implement the XYZ operand."""` without any reference to `__eq__`?

Comment (peterjc) [inline Bio/Seq.py]: That seems like an improvement.

File: Bio/Seq.py (hunk 6 above)
Comment (Gasta88) [inline Bio/Seq.py]: The function itself does not make sense.

How about `Return the reverse complement of an unknown nucleotide,which equals itself.`?

Comment (peterjc) [inline Bio/Seq.py]: If you think the function does not make sense, then clearly the current docstring is not clear enough. How about:

```python
        def reverse_complement(self):
            """Return the reverse complement of an unknown sequence.

            The reverse complement of an unknown nucleotide equals itself:

            >>> from Bio.Seq import UnknownSeq
            >>> from Bio.Alphabet import generic_dna
            >>> example = UnknownSeq(6, generic_dna)
            >>> print(example)
            NNNNNN
            >>> print(example.reverse_complement())
            NNNNNN
            """
```

Comment (Gasta88) [inline Bio/Seq.py]: Much better, thanks.

**Comment by Gasta88** [comment]
File edited.

File: Bio/Seq.py [hunk 10]
```diff
@@ -1391,25 +1394,25 @@ def complement(self):
         return self
 
     def reverse_complement(self):
-        """The reverse complement of an unknown nucleotide equals itself.
+        """Return the reverse complement of an unknown sequence.
 
-        >>> my_nuc = UnknownSeq(10)
-        >>> my_nuc
```
Comment (peterjc) [inline Bio/Seq.py]: Is there a tab here? The display on GitHub looks different for some reason...

Comment (peterjc) [inline Bio/Seq.py]: Well, if it was a tab it is being removed in this change so never mind.

**Pull request merged**